src/Bicep.Core/Registry/TemplateSpecModuleRegistry.cs

```diff
@@ -41,8 +41,12 @@ public TemplateSpecModuleRegistry(IFileResolver fileResolver, ITemplateSpecRepos
 
         public override RegistryCapabilities GetCapabilities(TemplateSpecModuleReference reference) => RegistryCapabilities.Default;
 
-        public override ResultWithDiagnostic<ArtifactReference> TryParseArtifactReference(string? aliasName, string reference)
+        public override ResultWithDiagnostic<ArtifactReference> TryParseArtifactReference(string? aliasName, ArtifactType artifactType, string reference)
         {
+            if (artifactType != ArtifactType.Module)
+            {
+                return new(x => x.UnsupportedArtifactType(typeof(TemplateSpecModuleRegistry).Name, artifactType));
```

Comment:
This is going to output something like `TemplateSpecModuleRegistry` which isn't neccessarily going to be meaningful to the end-user. I would recommend instead hand-writing a string.